cobench/cobench.ml

```diff
@@ -83,3 +84,55 @@ module Remote = struct
     | `Bool false, `String msg -> Lwt.fail_with msg
     | _ -> Lwt.fail_with (J.to_string json)
 end
+
+let pr_bench value =
+  Format.printf
+    {|{"results": [{"name": "%s", "metrics": [{"name": "%s", "value": %f, "units": "ms"}]}]}@.|}
+    value
+
+let time (f : unit -> 'a) =
+  let t = Sys.time () in
+  let _ = f () in
+  let timing = Sys.time () -. t in
+  timing *. 1000.
+
+let test_bench f = Test.make ~name:"" (Staged.stage @@ f)
+
+let benchmark quota f test_name metric_name =
+  let ols =
+    Analyze.ols ~bootstrap:0 ~r_square:true ~predictors:Measure.[| run |]
+  in
+  let instances = Toolkit.Instance.[ monotonic_clock ] in
+  let cfg = Benchmark.cfg ~start:1000 ~quota:(Time.second quota) () in
+  let raw_results =
+    Benchmark.all cfg instances
+      (Test.make_grouped ~name:"benchmark" ~fmt:"%s %s" [ test_bench f ])
+  in
+  let results =
+    List.map (fun instance -> Analyze.all ols instance raw_results) instances
+  in
+  let results = Analyze.merge ols instances results in
+  let timings = Hashtbl.find results "monotonic-clock" in
+  Hashtbl.iter
+    (fun _ v ->
+      match Analyze.OLS.estimates v with
+      | None -> ()
+      | Some ts ->
+          let t = List.hd ts in
+          pr_bench test_name metric_name (t /. 1000000.))
+    timings;
+  ()
+
+let time_loop ?(quota = 1.) initial_timing test_name metric_name f =
+  let t_time = Sys.time () in
+  while Sys.time () -. t_time +. initial_timing < quota do
```

Comment:
Nice! I would prefer if the function `time_loop` didn't take the new argument `initial_timing` (because we can do without it ^^). This while condition is the same as:

```ocaml
while Sys.time () -. t_time < quota -. initial_timing do
```

So perhaps the remaining `quota` could be adjusted at some point to take into account the `initial_timing` :)